jax/experimental/jax2tf/jax2tf.py

```diff
@@ -243,7 +243,10 @@ def __init__(self, trace: 'TensorFlowTrace', val: TfValOrUnit):
     else:  # Must be a numeric value
       assert core.skip_checks or _is_tfval(val), f"Non TfVal: {val}"
       aval = xla.abstractify(val)  # type: ignore
-      self.val = tf.convert_to_tensor(np.array(val, aval.dtype), dtype=aval.dtype)  # type: ignore
+      source_array_dtype = aval.dtype
+      if aval.dtype == jnp.bfloat16:
+        source_array_dtype = jnp.float32
+      self.val = tf.convert_to_tensor(np.array(val, source_array_dtype), dtype=to_tf_dtype(aval.dtype))  # type: ignore
```

Comment:
I am curious why does this work?

Reply:
afaict, the issue is a combination of two bugs:
1) `aval.dtype` is actually `jnp.bfloat16`, which needed to be casted using `to_tf_dtype`, and it is what caused the error message shown in the referenced issue
2) `tf.convert_to_tensor` does not seem to have a conversion rule for a `np.array` with a custom type (which makes sense, and `jnp.bfloat16` is a custom type that does not exist in NumPy); once you fix 1), you get the following error if you try to run the repro code:
```python
ValueError: Failed to convert a NumPy array to a Tensor (Unsupported numpy type: NPY_USERDEF).
```

Reply:
As we discussed online we are going to be looking for a fix with higher coverage of the bfloat16 issues.